### PR TITLE
37 delete user account

### LINKED ISSUE #37: Delete user account
**Short description:**
Add the possibility to delete his own account and all the linked personal data. 

**Describe the solution you'd like**
Create an api call to remove all the personal data and account.

Create a Flutter screen which ask the REST API to delete the connected user data.

### PR DESCRIPTION
**Pull request by LucasTesnier**
# Purpose

The purpose of this pull request is to add the possibility of deleting his own account.

Fixes #37 

## Added features

* A new API call useful for deleting his own account have been implemented
 * Secured by Token auth
 * E-mail confirmation

* A new Flutter Screen have been implemented : Settings
 * Accessible after login
 * Button to delete account

## Updated features

* Add DateTime parsing for all the Json factory.

* Rebase the old credentials gestion in flutter by the new one with UserData class object.

* Add a new User field inside Prisma.schema for E-mail confirmation operation (confirmProcess -> String -> default "")

### CODE REVIEW

**Review by tbellicha — APPROVED** [review]

**Review by tbellicha — APPROVED** [review]
LGTM

File: server/server_app.js [hunk 1]
```diff
@@ -105,14 +105,16 @@ app.post('/api/signup', (req, res, next) => {
...
-      'Thank you for you registration to our service !\nPlease go to the following link to confirm your mail : http://localhost:8080/api/mail/verification?token=' +
-        token
-    ).catch((_error) => {
-      return res.status(401).send('Invalid e-mail address.')
-    })
+    gmail
+      .sendEmail(
+        user.email,
+        'Email Verification',
+        'Thank you for you registration to our service !\nPlease go to the following link to confirm your mail : http://localhost:8080/api/mail/verification?token=' +
```
Comment (GuyomT) [inline server/server_app.js:112]: just a question, do we let localhost by default ? Or precise the IP (and maybe the port) from a constant variable ?

File: server/server_app.js [hunk 2]
```diff
@@ -176,15 +178,72 @@ app.get('/api/mail/verification', async (req, res) => {
...
+})
+
+/**
+ * Get request to delete an account
+ * Send a confirmation e-mail before deleting.
  * Need to be authentified with a token.
  */
 app.get(
-  '/profile',
+  '/api/user/deleteAccount',
```
Comment (GuyomT) [inline server/server_app.js:223]: is the authentification service part of the Api ?

**Review by GuyomT — APPROVED** [review]
LGTM but questions in comments

File: server/server_app.js (hunk 2 above)
Comment (LucasTesnier) [reply server/server_app.js:223]: Yes, all user operation is a path from the API cause it interact with the Database.
But we will rebase it later with a more formal path norme. (Cf the list you made last week) ;)

File: server/server_app.js (hunk 1 above)
Comment (LucasTesnier) [reply server/server_app.js:112]: For the moment i let localhost:8080 to make the link clickable more faster in development phase.
We will rebase it when we professionalize the application (Part two or last MVP modification)

Comment (GuyomT) [reply server/server_app.js:112]: Perfect

**Review by ZQUEMA — APPROVED** [review]

**Pull request merged**